docs/fee_design.md

```diff
@@ -0,0 +1,173 @@
+
+Given this, the block verification should assert that all fee transactions in the block have the same transaction type No matter which fee tx types (1 or 2), it should calculate the fee balance of all transaction within the block. The only difference is how the fee claiming output can be spent.
+
+Child chain and watcher could even further deprecate old fee tx types afterward if not necessary anymore. The logic change could be done by upgrading child chain service and watcher together.
+
+## Adding fee exit feature to the Plasma Framework
+
+This section will discuss the how do we add the fee exit feature to our Plasma Framework using the chosen design, assuming we are launching the network without fee exit feature at the beginning.
+
+Since the network would be first running with a Payment transaction type which does not support having fee transaction type as input, to enable spending fee transaction into Payment transaction, we would need a Payment transaction v2 for that.
+
+
+As a result, a high level steps of adding fee exit feature would be:
+1. Implement the contracts to enable spending Fee transaction in Payment transaction V2
+2. Implements the child chain and watcher logic to support new transaction types: fee transaction type and Payment V2. Need to be able to check the correctness of the transactions, including special logic for fee transaction and the changes of block verification logics.
+3. Implement child chain to automatically generate fee transactions within each block.
+4. Makes sure that clients update to new watchers
+5. Could turn on the auto fee generation feature once all clients update to the new watcher
+6. Deploy the PaymentExitGame for Payment transaction v2 that has the ability to spend the fee transaction
+7. (Optional) update deposit verifier to use Payment transaction v2 directly
+8. Wait 3 weeks for the new ExitGame contract to take effect
+9. Spend fee transaction in Payment transaction v2 and exit.
+
+## POC
+
+* Fee transaction type that have special nonce output and would be spent by Payment transaction and exit via Payment transaction: https://github.com/omisego/plasma-contracts/commit/7e6bf7a5886a5b1131d5346a71e923a64fab4b38
+
+## Reference
+
+* Tetsuji blockchain fee design: https://github.com/omisego/elixir-omg/blob/master/docs/tesuji_blockchain_design.md#fees
+* Scope of Five Guys
+* Fee exit high level discussion: https://github.com/omisego/plasma-contracts/issues/165
+
+## FAQ
+
+Q1: Is it possible for the operator to start collecting fee now, but defer adding the new fee-exit ALD to some time in the future? 
+> Yes, it should be possible in the abandoned design. But be aware pre-collecting fee meanings we would need to put extra effort to migrate DB for fee feature during production on live. For instance, we would need to calculate the sum of fee while accepting new transactions, which might impact sum of fee too.
```

Comment:
Mentions `abandoned design` but it's no longer present in the doc. We need to rephrase the answer?